### DISCUSSION TITLE
[Keyball61]トラックボール無し基盤のいくつかのキーが反応しない

### DISCUSSION THREAD

**Discussion by sota1235** [Yowkees/keyball#328] [Q&A] [answered]
Keyball61で左手をトラックボールなしの基盤ではんだづけしています。
その際に以下の写真で丸をつけたキーが反応しません。

<img width="963" alt="image" src="https://github.com/Yowkees/keyball/assets/2968860/285dbb29-6e16-46af-9a72-e4c40f0c7ba5">

丸無しの写真も置いておきます。

![image](https://github.com/Yowkees/keyball/assets/2968860/fad467d2-3331-4337-834f-4dc7c75d5343)

以下は試しました

- 該当のキースイッチのはんだ付けのし直し
- 周辺で怪しいダイオードのはんだ付けのし直し

いくつかをはんだ付けし直したところ、動いてたところが動かなくなったりしたのでどこにフォーカスして修正すべきか教えていただけると幸いです…。
お手数おかけしますがよろしくお願いします。

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

不具合キーは縦一列や横一列のキーがすべて動かないという感じではなく分散していますので、やはり該当のキーソケットとダイオードの半田付けを根気よく見直してください。特にキーソケットはとてもテンプラしやすいので、端子を横から覗いて、端子と基板の隙間に半田が流れているかご確認ください。
なお、該当するダイオードは下記のものになります。
![KB61_5keyfix](https://github.com/Yowkees/keyball/assets/84209772/658842bb-076e-4454-8e91-75011037b3cf)

また、キーソケット自体に問題がある場合もあります。下記画像のように金属端子がひらいて導通不良になる場合がありますので、こちらもご確認ください。キーソケットの両端子をピンセットでショートすれば入力できる場合はこの原因の可能性があります。
![Keysocket_NG](https://github.com/Yowkees/keyball/assets/84209772/6fbc2be5-bb81-4a44-a50d-8c8739b6b1cc)

以上、お手数ですがご確認お願いいたします。

**Reply by sota1235** [reply]
遅れましたがアドバイス通りに1つ1つ、ダイオードの半田付けをしなおしたところ全てのキーが動作しました！
左右の接続も問題なかったのでこれでKeyballデビューできます。

ご丁寧にご回答いただきありがとうございました 🙇
